ftw/referencewidget/browser/generate_pathbar.py

```diff
@@ -0,0 +1,24 @@
+from Acquisition import aq_parent
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+from Products.Five import BrowserView
+import json
+
+
+class GeneratePathbar(BrowserView):
+
+    def __call__(self):
+        originpoint = self.request.get('origin', None)
+        if not originpoint:
+            obj = self.context.context
+        else:
+            obj = self.context.context.unrestrictedTraverse(
+                originpoint.encode("utf8"))
+        results = []
+        while True:
+            results.insert(0, {'title': obj.Title(),
```

Comment:
This breaks when I have no `View` permission on the parent. Please make it safe: the title can be displayed in this situation (matching plone default behavior (pathbar) as well as other places).
You may need to make an unrestricted access.